spock-core/src/main/java/org/spockframework/runtime/DataIteratorFactory.java

```diff
@@ -93,6 +93,29 @@ protected int estimateNumIterations(Object[] dataProviders) {
       return result == Integer.MAX_VALUE ? UNKNOWN_ITERATIONS : result;
     }
 
+    protected int estimateNumIterations(IDataIterator[] dataProviderIterators) {
```

Comment:
Is the other method `estimateNumIterations(Object[] dataProviders)` then still be in use?

It seams a bit duplicated to the new one.

Reply:
It is quite similar yes.
But the other method is still used, yes.
For example in the `DataProviderMultiplier`.
This method does roughly the same as the other method but uses the already estimated amounts in the `IDataIterator`s instead of reestimating.

Reply:
Your comment inspired me, so I rewrote it a bit and thus invalidated your review, please have another look :-)